src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -182,9 +177,67 @@ void printFromInputStream(InputStream inputStream) throws IOException {
         }
     }
 
-    static String getNativeImagePath (ProjectConfiguration configuration) {
+    private static String getNativeImagePath (ProjectConfiguration configuration) {
         String graalPath = configuration.getGraalPath();
         Path path = Path.of(graalPath, "bin", "native-image");
         return path.toString();
     }
+
+    private Process startAppProcess( Path appPath, String appName ) throws IOException {
+        ProcessBuilder runBuilder = new ProcessBuilder(appPath.resolve(appName).toString());
+        runBuilder.redirectErrorStream(true);
+        return runBuilder.start();
+    }
+
+    public boolean compileAdditionalSources(ProcessPaths paths, ProjectConfiguration projectConfiguration)
+            throws IOException, InterruptedException {
+
+        String appName = projectConfiguration.getAppName();
+        Path workDir = paths.getGvmPath().resolve(appName);
+        Files.createDirectories(workDir);
+
+        ProcessBuilder processBuilder = new ProcessBuilder("gcc");
+        processBuilder.command().add("-c");
+        if (projectConfiguration.isVerbose()) {
+            processBuilder.command().add("-DGVM_VERBOSE");
+        }
+
+        for( String res: C_RESOURCES ) {
+            FileOps.copyResource("/native/linux/" + res, workDir.resolve(res));
```

Comment:
this is linux specific??

Reply:
The reason I did it is maintainability.  The code was identical for both Linux and Mac. One place to fix is much better than two. If MacOS code starts to differ, specialized method can be created.

Reply:
So far, same files are used for both linux and mac os. So we probably should rename the folder to `posix` (as Johan suggested). I doubt this will work for other platforms, so this will probably will change soon. For now it is okay.